File: api/health.go
package handler

import (
	"net/http"
	"time"

	"github-player-rating/internal/httpx"
)

// Handler is the Vercel Go entrypoint for GET /api/health.
func Handler(w http.ResponseWriter, r *http.Request) {
	if httpx.HandlePreflight(w, r) {
		return
	}
	httpx.JSON(w, http.StatusOK, map[string]interface{}{
		"status": "ok",
		"time":   time.Now().UTC().Format(time.RFC3339),
	})
}

File: api/health.ts
import type { VercelRequest, VercelResponse } from "@vercel/node";
import { handler as healthHandler } from "../server/src/health.js";

/**
 * Vercel maps this file to GET /api/health automatically.
 *
 * IMPORTANT: /api must contain ONLY these thin entrypoint files. Vercel turns every
 * .ts/.js/.mjs file anywhere under /api into its own serverless function, and the
 * Hobby plan caps a deployment at 12. All real logic lives in /server, outside /api.
 */
export default async function handler(req: VercelRequest, res: VercelResponse): Promise<void> {
  try {
    await healthHandler(req, res);
  } catch (err) {
    console.error("health error:", err);
    if (!res.headersSent) {
      res.status(500).json({ error: "internal_error", message: "something went wrong" });
    }
  }
}
